### ISSUE TITLE
tailwind_axum example broken when trying to set LEPTOS_SITE_PKG_DIR env var for cache busting

### ISSUE THREAD

**Issue by wt** [leptos-rs/leptos#2004]
**Describe the bug**
In the tailwind_axum example, setting the env var `LEPTOS_SITE_PKG_DIR` to anything other than "pkg" causes site to fail at runtime with errors on JS console.

**Leptos Dependencies**

Head revision of the main branch.

**To Reproduce**
1. Change to root of repo
2. cd examples/tailwind_axum
3. LEPTOS_SITE_PKG_DIR=blah cargo leptos watch
4. after service is started, open http://localhost:3000 in browser
5. CSS doesn't load as the app is trying to load it from /pkg/tailwind.css (see app.rs)

**Expected behavior**
The CSS is loaded and works.

**Additional context**
If I manually override the setting the App object to load the css from the right location, I then get other errors in the js console:

    tailwind.js:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.

FWIW, I was experimenting with trying to implement cache busting by generating a random string to insert into the LEPTOS_SITE_PKG_DIR. Also, it seems very add that the App doesn't seem to know where to the load the CSS from if it isn't the default. I am not sure how to fix this. Should the config be plumbed into the App? Is it already available somehow?

**Comment by gbj** [comment]
I don't see a step where you changed
```
        <Stylesheet id="leptos" href="/pkg/tailwind.css"/>
```
to
```
        <Stylesheet id="leptos" href="/blah/tailwind.css"/>
```
If I do the rest of your steps and point the `Stylesheet` component to the correct URL, it works as expected.

> Also, it seems very add that the App doesn't seem to know where to the load the CSS from if it isn't the default. I am not sure how to fix this. Should the config be plumbed into the App? Is it already available somehow?
Not sure if there's a typo or something, but I can't figure out what this means, sorry.

**Comment by wt** [comment]
Let me try again.

Running something like `LEPTOS_SITE_PKG_DIR=blah cargo leptos watch` should not require someone to change the source of the App component to match in order for the site to work.

Is there some way for the App component to know what the pkg dir is so that the stylesheet path doesn't need to be changed for that to work? The app should just know that the stylesheet needs to be sourced from the appropriate dir.

So, I looked to see if I could somehow load the config in the component. That doesn't appear to be possible since the component needs to run in the wasm environment. Is there some other way to discover the pkg dir?

**Comment by wt** [comment]
Maybe a way to see the config at compile time would help here? Maybe via a proc macro of some kind? This makes sense in my mind if these are literally compile time options. Then the option value could be locked in at compile time.

**Comment by gbj** [comment]
Ok, I think I understand and just disagree, honestly. The `<Stylesheet/>` component there isn't magic, it's just injecting a `<link rel="stylesheet">` into the `<head>`. It knows nothing about `cargo-leptos` and it's the config options it uses, because while `cargo-leptos` is great, it's not required to use the framework. 

You could certainly do something with [the `env!()` macro](https://doc.rust-lang.org/std/macro.env.html) to generate the URL of the stylesheet.

But I don't think it's unreasonable to say that if you change one part of the application you might need to change another part of the application to match. i.e., the starter templates work as they are, and if you change `"pkg"` in one place you might need to look for other references to `pkg` and change those too.

**Comment by wt** [comment]
The problem isn't changing different parts of the code to reflect a change. The problem is capturing a compile time option that is set by an env var can could change in future builds.

For example, if I didn't have to change code to capture the pkg directory, I could use LEPTOS_SITE_PKG_DIR to pick a randomly named directory to implement cache busting.

As it stands, I am not quite sure how to easily implement a cache busting scheme starting with leptos+axum starter.

**Comment by gbj** [comment]
Thanks, that clarifies the use case for me. How about the approach using the `env!` macro from `std` I suggested in my last reply? As far as I can tell this works:
```
<Stylesheet href=format!("{}/leptos_start.css", env!("LEPTOS_SITE_PKG_DIR"))/>
```
Seems to work fine for me. (I am playing with an Actix template and had to update the file serving to make sure it used that same env variable for its static file serving path, I can't remember how it's set up in the Axum example you're working with and it may just work)

**Comment by wt** [comment]
I may be able to do something with the `env!` macro. I want to acknowledge that.

I'm not super stoked about parsing the value of the var separately from leptos since this introduces the possibility of doing is slightly differently, but maybe it's simple enough.

Also, I do believe you're example may work. I'll give it a try when I have some time.

Also, it really is odd to me that i can neither access build config, like the pkg dir setting, from the wasm environment nor can I pass in the config and use it in the root component.

I'm used to something like Next.js that allows capturing certain config so that it can be used in the wasm environment. Like described [here](https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables). I don't know that I would say this is the best design, but it is very useful sometimes.

Having said all this, I do really like what I've seen of Leptos so far.

**Comment by wt** [comment]
I got this to work:

```
diff --git a/examples/tailwind_axum/src/app.rs b/examples/tailwind_axum/src/app.rs
index 7ebc2aad..545ba6c4 100644
--- a/examples/tailwind_axum/src/app.rs
+++ b/examples/tailwind_axum/src/app.rs
@@ -6,9 +6,11 @@ use leptos_router::*;
 pub fn App() -> impl IntoView {
     provide_meta_context();
 
+    let site_pkg_dir = option_env!("LEPTOS_SITE_PKG_DIR").unwrap_or("pkg");
+
     view! {
 
-        <Stylesheet id="leptos" href="/pkg/tailwind.css"/>
+        <Stylesheet id="leptos" href=format!("/{}/tailwind.css", site_pkg_dir)/>
         <Link rel="shortcut icon" type_="image/ico" href="/favicon.ico"/>
         <Router>
             <Routes>
```

I will say that that is better. However, I still think it would be better to have some way to access this config from the wasm side.
